### PR TITLE
Added fields as query parameter

### PR DESCRIPTION
**Pull request by PhyberApex**
This is really a first throw of me to check if you are okay with this or if you have anything against it. I also fixed all tests and added one more test. As I am not that fluent in C# and WebAPI tho I can't really tell if this is enough.

Be sure to get back to me with any comments you have and what you need changed to make a PR possible.

~Cheers

### CODE REVIEW

**Comment by PhyberApex** [comment]
Also I have no idea why that one test failed. It failed about 2 times on my machine as well but now it is working fine.

File: Tests/JsonApiSerializerTests.cs [hunk 1]
```diff
@@ -68,6 +68,22 @@ public void ConditionallyAppliesFilters()
 
         }
 
+		[Fact(DisplayName = "Uses query fieldset expressions if specified")]
```
Comment (joukevandermaas) [inline Tests/JsonApiSerializerTests.cs]: I think we need a test where we ask for a field that's not on the model (should get nothing in that case).

File: Saule/Queries/Fieldset/FieldsetProperty.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+        /// <summary>
+        /// Gets property fields
+        /// </summary>
+        public string[] Fields { get; }
+
+        /// <inheritdoc/>
+        public override string ToString()
+        {
+            return $"filter[{Type}]={string.Join(",", Fields)}";
```
[the remark this answers is not in the document]
Comment (PhyberApex) [reply Saule/Queries/Fieldset/FieldsetProperty.cs]: Fixed in #199

File: Tests/JsonApiSerializerTests.cs [hunk 3]
```diff
@@ -68,6 +68,22 @@ public void ConditionallyAppliesFilters()
...
+        public void UsesQueryFieldsetExpressions()
```
[the remark this answers is not in the document]
Comment (PhyberApex) [reply Tests/JsonApiSerializerTests.cs]: Having some trouble detecting those on my machine I hope they got fixed with a0ee8584e64fe2562d16f05a175a6225962dee00

File: Tests/JsonApiSerializerTests.cs (hunk 1 above)
Comment (PhyberApex) [reply Tests/JsonApiSerializerTests.cs]: Added the test in a0ee8584e64fe2562d16f05a175a6225962dee00. I think only the attributes should be empty as id and type field are still returned which I think would be correct behaviour at this point.

**Comment by PhyberApex** [comment]
I guess the pre-release is held up by the failing test? Should I take a look at that as well?

~Cheers

**Comment by joukevandermaas** [comment]
@PhyberApex you're right; but it's nothing to do with your PR. I'll fix it and publish the release. Thanks!

**Comment by PhyberApex** [comment]
Hey @joukevandermaas sorry to bother you again. Just realized that there does not seem to be a release yet. Any estimate on this one? Would be greatly appriciated! Thanks!

~Cheers

**Comment by joukevandermaas** [comment]
@PhyberApex it is indeed a beta release. The betas are usually pretty stable; in this specific case I think it *only* contains the changes from this PR (compared to stable). Before creating a stable release I'll need to (at least) add some docs for this feature.

**Pull request merged**